tests/lib/tools/system-state

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+show_help() {
+    echo "usage: is-core"
+    echo "       is-core-16"
+    echo "       is-core-18"
+    echo "       is-core-20"
+    echo "       is-classic"
+    echo "       is-trusty"
+    echo "       is-xenial"
+    echo "       is-bionic"
+    echo "       is-focal"
+    echo ""
+    echo "Get general information about the current system"
+}
+
+is_core(){
+    grep "NAME=\"Ubuntu Core\"" /etc/os-release
```

Comment:
Neat idea.

I'd change this slightly as follows:
```suggestion
    grep -qFx 'NAME="Ubuntu Core"' /etc/os-release
```

- `-q` will refrain from printing the match
- `-F` will match fixed strings, not regular expressions (excellent here since we don't have to worry about quoting)
- `-x` will automatically match entire lines only, avoiding the risk of matching a partial line by mistake

Reply:
done